### ISSUE TITLE
Add support for asynch iters

### ISSUE THREAD

**Issue by DrChainsaw** [DrChainsaw/AmpControl#27]
Motivation is that with more than one input channel, the data sets of the validation set tend to not fit in RAM unless a very small evaluation set is used.

Furthermore, the current way of caching the DataSets in RAM is kinda shaky and tends to break with updates to dl4j and is simply not the recommended way of doing it.

Main hurdle is that all random data augmentation etc needs to be "resetable" from iterators to ensure that 1) validation set is always identical between evaluations and 2) training data for different models is always identical.
